app/controllers/articles_controller.rb

```diff
@@ -6,4 +6,29 @@ def index
     def show
         @article = Article.find(params[:id])
     end
+
+    def new
+        @article = Article.new
+        @categories = Category.all
+    end
+
+    def create
+        @article = Article.new(article_params)
+        @article.update(category: find_category)
+        if @article.save
+            redirect_to article_path(@article)
+        else
+            redirect_to new_article_path, notice: "Every field needs to be filled in!"
+        end
+    end
+
+    private
+    def article_params
+        params.require(:article).permit(:title, :description, :content, :journalist)
+    end
+
+    def find_category
+        Category.find_by(id: params[:article][:category_id])
```

Comment:
No need to use `find_by` This can be simplified to:

```ruby
Category.find(params[:article][:category_id])
```

Reply:
Done